src/release/breaking-changes/material-3-migration.md

```diff
@@ -0,0 +1,515 @@
+
+Code before migration:
+
+```dart
+ElevatedButton(
+  onPressed: () {},
+  child: const Text('Button'),
+),
+```
+
+Code after migration:
+
+```dart
+ElevatedButton(
+  style: ElevatedButton.styleFrom(
+    backgroundColor: Theme.of(context).colorScheme.primary,
+    foregroundColor: Theme.of(context).colorScheme.onPrimary,
+  ),
+  onPressed: () {},
+  child: const Text('Button'),
+),
+```
+
+### Typography
+
+The default values for `ThemeData.textTheme` are updated to match the
+Material 3 defaults. Changes include updated font size, font weight, letter
+spacing, and line height. For more details, check out the [`TextTheme`][]
+documentation.
+
+As shown in the following example, prior to the 3.16 release, when a `Text`
+widget with a long string using `TextTheme.bodyLarge` in a constrained layout
+wrapped the text into two lines. However, the 3.16 release wraps the text into
+three lines. If you must achieve the previous behavior, adjust the text style
+and, if necessary, the letter spacing.
+
+Code before migration:
+
+```dart
+ConstrainedBox(
+  constraints: const BoxConstraints(maxWidth: 200),
+    child: Text(
+      'This is a very long text that should wrap to multiple lines.',
+      style: Theme.of(context).textTheme.bodyLarge,
+  ),
+),
+```
+
+Code after migration:
+
+```dart
+ConstrainedBox(
+  constraints: const BoxConstraints(maxWidth: 200),
+    child: Text(
+      'This is a very long text that should wrap to multiple lines.',
+      style: Theme.of(context).textTheme.bodyMedium!.copyWith(
+        letterSpacing: 0.0,
+      ),
+  ),
+),
+```
+
+[`TextTheme`]: {{site.api}}/flutter/material/TextTheme-class.html
+
+### Components
+
+Some components couldn't merely be updated to match the Material 3 Design spec
+but needed a whole new implementation. Such components require manual migration
+since the Flutter SDK doesn’t know what, exactly, you want.
+
+Replace the Material 2 style [`BottomNavigationBar`][] widget with the new
+[`NavigationBar`][] widget. It’s slightly taller, contains pill-shaped
+navigation indicators, and uses new color mappings.
+
+Code before migration:
+
+```dart
+BottomNavigationBar(
+  items: const <BottomNavigationBarItem>[
+    BottomNavigationBarItem(
+      icon: Icon(Icons.home),
+      label: 'Home',
+    ),
+    BottomNavigationBarItem(
+      icon: Icon(Icons.business),
+      label: 'Business',
+    ),
+    BottomNavigationBarItem(
+      icon: Icon(Icons.school),
+      label: 'School',
+    ),
+  ],
+),
+```
+
+Code after migration:
+
+```dart
+NavigationBar(
+  destinations: const <Widget>[
+    NavigationDestination(
+      icon: Icon(Icons.home),
+      label: 'Home',
+    ),
+    NavigationDestination(
+      icon: Icon(Icons.business),
+      label: 'Business',
+    ),
+    NavigationDestination(
+      icon: Icon(Icons.school),
+      label: 'School',
+    ),
+  ],
+),
+```
+
+Check out the complete sample on
+[migrating from BottomNavigationBar from NavigationBar][].
```

Comment:
There are two "froms" here. Consider something like:

```
For more guidance, check out another sample of
[migrating from `BottomNavigationBar` to `NavigationBar`][].
```

(Do upgrade the below link definition as well to whatever you decide :)